src/hipscat/io/file_io/file_io.py

```diff
@@ -23,126 +26,329 @@ def make_directory(file_pointer: FilePointer, exist_ok: bool = False):
     """
-    pq.write_metadata(schema, file_pointer, metadata_collector=metadata_collector, **kwargs)
 
+    file_system, file_pointer = get_fs(file_pointer=file_pointer, storage_options=storage_options)
 
-def read_fits_image(map_file_pointer: FilePointer):
+    file_pointer = strip_leading_slash_for_pyarrow(file_pointer, protocol=file_system.protocol)
+    pq.write_metadata(
+        schema, file_pointer, metadata_collector=metadata_collector, filesystem=file_system, **kwargs
+    )
+
+
+def read_fits_image(map_file_pointer: FilePointer, storage_options: dict = {}):
     """Read the object spatial distribution information from a healpix FITS file.
 
     Args:
         file_pointer: location of file to be written
+        storage_options: dictionary that contains abstract filesystem credentials
     Returns:
         histogram (:obj:`np.ndarray`): one-dimensional numpy array of long integers where the
             value at each index corresponds to the number of objects found at the healpix pixel.
     """
-    return hp.read_map(map_file_pointer)
+    file_system, map_file_pointer = get_fs(file_pointer=map_file_pointer, storage_options=storage_options)
+    with tempfile.NamedTemporaryFile() as _tmp_file:
+        with file_system.open(map_file_pointer, "rb") as _map_file:
+            map_data = _map_file.read()
+            _tmp_file.write(map_data)
+            map_fits_image = hp.read_map(_tmp_file.name)
+    return map_fits_image
 
 
-def write_fits_image(histogram: np.ndarray, map_file_pointer: FilePointer):
+def write_fits_image(histogram: np.ndarray, map_file_pointer: FilePointer, storage_options: dict = {}):
     """Write the object spatial distribution information to a healpix FITS file.
 
     Args:
         histogram (:obj:`np.ndarray`): one-dimensional numpy array of long integers where the
             value at each index corresponds to the number of objects found at the healpix pixel.
         file_pointer: location of file to be written
+        storage_options: dictionary that contains abstract filesystem credentials
+    """
+    file_system, map_file_pointer = get_fs(file_pointer=map_file_pointer, storage_options=storage_options)
+    with tempfile.NamedTemporaryFile() as _tmp_file:
+        with file_system.open(map_file_pointer, "wb") as _map_file:
+            hp.write_map(_tmp_file.name, histogram, overwrite=True, dtype=np.int64)
+            _map_file.write(_tmp_file.read())
+
+
+def read_yaml(file_handle: FilePointer, storage_options: dict = {}):
+    """Reads yaml file from filesystem.
+
+    Args:
+        file_handle: location of yaml file
+        storage_options: dictionary that contains abstract filesystem credentials
+    """
+    file_system, file_handle = get_fs(file_pointer=file_handle, storage_options=storage_options)
+    with file_system.open(file_handle, "r", encoding="utf-8") as _file:
+        metadata = yaml.safe_load(_file)
+    return metadata
+
+def copy_tree_fs_to_fs(
```

Comment:
If this is only called from tests, I'm not sure we need to expose it here in the public API.

Reply:
I've made it a pytest fixture and taken it out of io